zetaclient/chains/evm/observer/observer_gas.go

```diff
@@ -0,0 +1,146 @@
+	}
+
+	// start gas price ticker
+	ticker, err := clienttypes.NewDynamicTicker(
+		fmt.Sprintf("EVM_WatchGasPrice_%d", ob.Chain().ChainId),
+		ob.GetChainParams().GasPriceTicker,
+	)
+	if err != nil {
+		ob.Logger().GasPrice.Error().Err(err).Msg("NewDynamicTicker error")
+		return err
+	}
+	ob.Logger().GasPrice.Info().Msgf("WatchGasPrice started for chain %d with interval %d",
+		ob.Chain().ChainId, ob.GetChainParams().GasPriceTicker)
+
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C():
+			if !ob.GetChainParams().IsSupported {
+				continue
+			}
+			err = ob.PostGasPrice(ctx)
+			if err != nil {
+				ob.Logger().GasPrice.Error().Err(err).Msgf("PostGasPrice error for chain %d", ob.Chain().ChainId)
+			}
+			ticker.UpdateInterval(ob.GetChainParams().GasPriceTicker, ob.Logger().GasPrice)
+		case <-ob.StopChannel():
+			ob.Logger().GasPrice.Info().Msg("WatchGasPrice stopped")
+			return nil
+		}
+	}
+}
+
+// PostGasPrice posts gas price to zetacore.
+func (ob *Observer) PostGasPrice(ctx context.Context) error {
+	// GAS PRICE
+	gasPrice, err := ob.evmClient.SuggestGasPrice(ctx)
+	if err != nil {
+		return errors.Wrap(err, "unable to suggest gas price")
+	}
+
+	// PRIORITY FEE (EIP-1559)
+	priorityFee, err := ob.determinePriorityFee(ctx)
+	if err != nil {
+		return errors.Wrap(err, "unable to determine priority fee")
+	}
+
+	blockNum, err := ob.evmClient.BlockNumber(ctx)
+	if err != nil {
+		return errors.Wrap(err, "unable to get block number")
+	}
+
+	_, err = ob.
+		ZetacoreClient().
+		PostVoteGasPrice(ctx, ob.Chain(), gasPrice.Uint64(), priorityFee.Uint64(), blockNum)
+
+	if err != nil {
+		return errors.Wrap(err, "unable to post vote for gas price")
+	}
+
+	return nil
+}
+
+// determinePriorityFee determines the chain priority fee.
+// Returns zero for non EIP-1559 (London fork) chains.
+func (ob *Observer) determinePriorityFee(ctx context.Context) (*big.Int, error) {
+	supported, err := ob.supportsPriorityFee(ctx)
+	switch {
+	case err != nil:
+		return nil, err
+	case !supported:
+		// noop
+		return big.NewInt(0), nil
+	}
+
+	fee, err := ob.evmClient.SuggestGasTipCap(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to suggest gas tip cap")
+	}
+
+	return fee, nil
+}
+
+// supportsPriorityFee checks if the chain supports EIP-1559 (London fork).
+// uses cache so actual RPC call is made only once.
+func (ob *Observer) supportsPriorityFee(ctx context.Context) (bool, error) {
+	// noop
+	if ob.priorityFeeConfig.checked {
+		return ob.priorityFeeConfig.supported, nil
+	}
+
+	baseFee, err := ob.getChainBaseFee(ctx)
+	if err != nil {
+		return false, errors.Wrap(err, "unable to get base fee")
+	}
+
+	ob.Logger().GasPrice.Info().
+		Int64("observer.chain_id", ob.Chain().ChainId).
+		Str("observer.base_fee", baseFee.String()).
+		Msg("Fetched base fee for chain")
+
+	// EIP-1559 is supported if base fee is not zero.
+	isSupported := baseFee.Uint64() > 0
```

Comment:
I think we should just follow ChainSafe's logic (similar to below). For example, Binance mainnet returns `0` baseFee but it still supports EIP-1559 transactions.

`isSupported := baseFee != nil`

Reply:
Yep, you're right. https://github.com/zeta-chain/node/pull/2483/commits/ccc3598ead10d16816f4adeadd320030e9ff31aa